scalyr_agent/scalyr_client.py

```diff
@@ -1474,13 +1474,17 @@ def __init__(
         if (attrs is not None or thread_id is not None) and base is not None:
             raise Exception("Cannot use both attrs/thread_id and base")
 
+        if self.__attrs is None:
+            self.__attrs = dict()
+
         self.__thread_id = None
+        self.__parent_event = None
```

Comment:
Add comment: `Used to get attributes from the parent (such as the logfile attributes) for this event.`